File: app/src/main/java/com/owncloud/android/authentication/AccountAuthenticatorService.java
/*
 * Nextcloud - Android Client
 *
 * SPDX-FileCopyrightText: 2011 Bartosz Przybylski <bart.p.pl@gmail.com>
 * SPDX-License-Identifier: GPL-2.0-only
 */
package com.owncloud.android.authentication;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;

public class AccountAuthenticatorService extends Service {

    private AccountAuthenticator mAuthenticator;

    @Override
    public void onCreate() {
        super.onCreate();
        mAuthenticator = new AccountAuthenticator(this);
    }

    @Override
    public IBinder onBind(Intent intent) {
        return mAuthenticator.getIBinder();
    }

}

File: app/src/main/java/com/owncloud/android/authentication/AccountAuthenticatorService.kt
/**
 * ownCloud Android client application
 *
 * Copyright (C) 2011  Bartek Przybylski
 * Copyright (C) 2015 ownCloud Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License version 2,
 * as published by the Free Software Foundation.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http:></http:>//www.gnu.org/licenses/>.
 *
 */
package com.owncloud.android.authentication

import android.app.Service
import android.content.Intent
import android.os.IBinder

class AccountAuthenticatorService : Service() {
    private var mAuthenticator: AccountAuthenticator? = null

    override fun onCreate() {
        super.onCreate()
        mAuthenticator = AccountAuthenticator(this)
    }

    override fun onBind(intent: Intent): IBinder? {
        return mAuthenticator?.iBinder
    }
}
